ADAL/src/ADAuthenticationErrorConverter.m

```diff
@@ -69,13 +72,12 @@ + (ADAuthenticationError *)ADAuthenticationErrorFromMSIDError:(NSError *)msidErr
         domain = s_errorDomainMapping[domain];
     }
     
-    //Map errorCode. Note that errorCode must be mapped together with domain
+    //Map errorCode
     NSInteger errorCode = msidError.code;
-    NSString *mapKey = [NSString stringWithFormat:@"%@|%ld", msidError.domain, (long)errorCode];
-    NSString *mapValue = s_errorCodeMapping[mapKey];
-    if (![NSString msidIsStringNilOrBlank:mapValue])
+    NSNumber *mappedErrorCode = s_errorCodeMapping[msidError.domain][NSNUMBER(msidError.code)];
```

Comment:
We should first first check if domain exists, then check for the code

Reply:
👍 +1: